From b96b7889527c9ccd22b56104b71e3bd762d009e4 Mon Sep 17 00:00:00 2001
From: Viktoria Maximova <viktoria.maksimova@intel.com>
Date: Tue, 9 May 2023 10:40:14 -0700
Subject: [PATCH 124/172] [DebugInfo] Fix translation of DebugSource Text
 argument (#2003)

Handle the case when we have `DebugInfoNone` for the Text argument
which is usually expected to be `OpString`.
---
 lib/SPIRV/SPIRVToLLVMDbgTran.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/SPIRV/SPIRVToLLVMDbgTran.cpp b/lib/SPIRV/SPIRVToLLVMDbgTran.cpp
index 760b0712..80a4f896 100644
--- a/lib/SPIRV/SPIRVToLLVMDbgTran.cpp
+++ b/lib/SPIRV/SPIRVToLLVMDbgTran.cpp
@@ -144,6 +144,8 @@ const std::string &SPIRVToLLVMDbgTran::getString(const SPIRVId Id) {
 Optional<std::string>
 SPIRVToLLVMDbgTran::getStringContinued(const SPIRVId Id,
                                        SPIRVExtInst *DebugInst) {
+  if (getDbgInst<SPIRVDebug::DebugInfoNone>(Id))
+    return std::string();
   std::string Str = BM->get<SPIRVString>(Id)->getStr();
   using namespace SPIRVDebug::Operand::SourceContinued;
   for (auto *I : DebugInst->getContinuedInstructions()) {
-- 
2.20.1

